### ISSUE TITLE
Hi-DPI scalling no longer working

### ISSUE THREAD

**Issue by samtygier** [flathub/org.telegram.desktop#138]
Since a recent update telegram no longer scales properly on my Hi-DPI screen and so is tiny and hard to use.

The first bad version is
```
Commit: d28a3f986040ddb4d62891dbbca67c21d470157f0c0a3ef009e76b2f74fa414e
   Subject: Add new release entry to appdata (18bc718d)
      Date: 2020-01-23 12:48:49 +0000
```

If I revert to:
```
Commit: e68d712c7addd1d0fc8871ba29d46c9a73de81c2eb3cec8cc57854f3208845ff
   Subject: Drop bundled ffmpeg and use .ffmpeg-full extension (ad5ae43a)
      Date: 2019-12-09 18:49:49 +0000
```
It renders and a sensible size.

I am using MATE desktop 1.22.2 on Fedora 30.

**Comment by nirbheek** [comment]
I can reproduce this on Fedora 31 with GNOME Shell too.

**Comment by grulja** [comment]
This is because Telegram is forcing to use xcb instead of wayland platform plugin. I will have to check what we can do about it, because some people report glitches on Wayland.

**Comment by nirbheek** [comment]
> This is because Telegram is forcing to use xcb instead of wayland platform plugin

Link for posterity: https://github.com/flathub/org.telegram.desktop/pull/134

**Comment by ilya-fedin** [comment]
> This is because Telegram is forcing to use xcb instead of wayland platform plugin.

Emm... MATE 1.22 doesn't support wayland. The most likely reason is https://github.com/telegramdesktop/tdesktop/commit/85acdbc7ed34167844c18006c7796889cb641735#diff-f0bf3e91862d6c3e1e312896c9dd7a63.
It was disabled by upstream as unsupported and, so, it is not a bug, since Telegram has the custom scaling feature. ¯\_(ツ)_/¯

**Comment by kowalski7cc** [comment]
Having the same issue on Fedora 32, Gnome 3.36.1. It seems Telegram-Flatpak runs inside Xwayland. No issues with Telegram from RpmFusion, which runs inside Wayland correctly.
When dragged to the second screen (which has 200% scaling) it will not scale.

**Comment by ilya-fedin** [comment]
Set scaling in app settings

**Comment by gasinvein** [comment]
@kowalski7cc If Telegram on Wayland works better for you, just unset the var that forces it to use xwayland, e.g. `flatpak override --user --env=QT_QPA_PLATFORM= org.telegram.desktop`

**Comment by kowalski7cc** [comment]
@gasinvein Perfect! Thanks for the tip :+1: 
I hope in the future Wayland will be as default :D

**Comment by ilya-fedin** [comment]
Is this still an issue?

**Comment by ilya-fedin** [comment]
Should be fixed by 938d47b76697cd95557f954175609529656cda33
